src/drivers/actuators/vertiq_io/vertiq_client_manager.hpp

```diff
@@ -0,0 +1,275 @@
+		}else{
+			PX4_INFO("Could not add this entry. Maximum number exceeded");
+		}
+	}
+
+private:
+	/**
+	* @brief Initialize all of the Vertiq Clients that we want to use
+	*
+	* @param object_id The object ID with which to initialize our clients
+	*/
+	void InitVertiqClients(uint8_t object_id);
+
+	/**
+	* @brief Initialize all of the Entry Wrappers
+	*/
+	void InitEntryWrappers();
+
+	uint8_t _object_id_now;
+
+	//We need a serial handler in order to talk over the serial port
+	VertiqSerialInterface *_serial_interface;
+
+	//These exist whenever we're using IQUART
+	EntryWrapper<float, float> _velocity_max_entry;
+	EntryWrapper<float, float> _voltage_max_entry;
+	EntryWrapper<uint8_t, int32_t> _control_mode_entry;
+	EntryWrapper<uint8_t, int32_t> _motor_direction_entry;
+	EntryWrapper<uint8_t, int32_t> _fc_direction_entry;
+
+#ifdef CONFIG_USE_IFCI_CONFIGURATION
+	EntryWrapper<uint8_t, int32_t> _throttle_cvi_entry;
+#endif //CONFIG_USE_IFCI_CONFIGURATION
+
+#ifdef CONFIG_USE_PULSING_CONFIGURATION
+	EntryWrapper<uint8_t, int32_t> _pulsing_voltage_mode_entry;
+	EntryWrapper<uint8_t, int32_t> _x_cvi_entry;
+	EntryWrapper<uint8_t, int32_t> _y_cvi_entry;
+	EntryWrapper<float, float> _pulse_zero_angle_entry;
+	EntryWrapper<float, float> _pulse_velo_cutoff_entry;
+	EntryWrapper<float, float> _pulse_torque_offset_entry;
+	EntryWrapper<float, float> _pulse_volt_limit_entry;
+#endif //CONFIG_USE_PULSING_CONFIGURATION
+
+////////////////////////////////////////////////////////////////////////
+//Vertiq client information
+	//Some constants to help us out
+	static const uint8_t MAXIMUM_CONFIGURATION_CLIENTS = 20; //These are clients whose module ID will change when Target Module ID changes
+	static const uint8_t MAXIMUM_OPERATIONAL_CLIENTS = 20; //These are clients that are used for module control/telemetry. They have a static Module ID
+
+	//Client Arrays
+	ClientAbstract *_configuration_client_array[MAXIMUM_CONFIGURATION_CLIENTS];
+	uint8_t _configuration_clients_in_use = 0;
+
+	ClientAbstract *_operational_client_array[MAXIMUM_OPERATIONAL_CLIENTS];
+	uint8_t _operational_clients_in_use = 0;
+
+	//Clients
+	PropellerMotorControlClient _broadcast_prop_motor_control;
+	ArmingHandlerClient _broadcast_arming_handler;
+	IQUartFlightControllerInterfaceClient *_telem_ifci;
+	EscPropellerInputParserClient *_prop_input_parser_client;
+
+#ifdef CONFIG_USE_IFCI_CONFIGURATION
+	//Make all of the clients that we need to talk to the IFCI config params
+	IQUartFlightControllerInterfaceClient *_ifci_client;
+#endif //CONFIG_USE_IFCI_CONFIGURATION
+
+#ifdef CONFIG_USE_PULSING_CONFIGURATION
+	VoltageSuperPositionClient *_voltage_superposition_client;
+	PulsingRectangularInputParserClient *_pulsing_rectangular_input_parser_client;
+#endif //CONFIG_USE_PULSING_CONFIGURATION
+////////////////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////////////////
+//Vertiq Client Entry information
```

Comment:
Not to harp on comments, but having some more here explaining what is going on with these lists, why they are necessary, and a bit of how I can use these would be helpful